Justfile

```diff
@@ -89,12 +89,12 @@ get-tag-version:
 
 # create the git tag from Cargo.toml, must be on main
 tag: ensure-main
```

Comment:
the only thing here that we might wanna safeguard against is to compare local HEAD vs remote HEAD cause sometimes folks can forget to pull the change that bumped up the version

Reply:
if that's not easy to do i wonder if we can print a msg, hit enter to continue...

Reply:
good idea, updated `ensure-main` to verify both that you're on `main` and you're on the latest `main` 
which also covers the `release version` case to makes sure you're cutting a release off the latest `main`

```bash
❯ just tag-push
Error: Your branch is not up to date with the upstream main branch
  ensure your branch is up to date (git pull)
error: Recipe `ensure-main` failed with exit code 1
```